inc/saiport.h

```diff
@@ -201,9 +201,53 @@ typedef enum _sai_port_ptp_mode_t
 
 } sai_port_ptp_mode_t;
 
+/**
+ * @brief Port Serdes Attribute ID Range
+ */
+#define SAI_PORT_ATTR_SERDES_RANGE 32
+
+/**
+ * @brief Attribute Id offset in the range of port serdes attribute
+ */
+typedef enum _sai_port_attr_offset_serdes_t
+{
+    /** Pre-emphasis offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_PREEMPHASIS = 0,
+
+    /** Idriver offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_IDRIVER = 1,
+
+    /** Ipredriver offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_IPREDRIVER = 2,
+
+    /** Tx fir Precursor1 tap-filter offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_TX_FIR_PRE1 = 3,
+
+    /** Tx fir Precursor2 tap-filter offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_TX_FIR_PRE2 = 4,
+
+    /** Tx fir Maincursor tap-filter offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_TX_FIR_MAIN = 5,
+
+    /** Tx fir Postcursor1 tap-filter offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_TX_FIR_POST1 = 6,
+
+    /** Tx fir Postcursor2 tap-filter offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_TX_FIR_POST2 = 7,
+
+    /** Tx fir Postcursor3 tap-filter offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_TX_FIR_POST3 = 8,
+
+    /** Max serdes attribute offset */
+    SAI_PORT_ATTR_OFFSET_SERDES_END = SAI_PORT_ATTR_SERDES_RANGE - 1
+
+} sai_port_attr_offset_serdes_t;
+
 /**
  * @brief Attribute Id in sai_set_port_attribute() and
  * sai_get_port_attribute() calls
+ *
+ * @flags Contains flags
```

Comment:
you changed this to be flags, and you added ranges for attributes, but none of those are defined, instead of adding min/max ranges, which at this point is incomplete, just add new attributes one by one, instead introducing incomplete ranges